### DISCUSSION TITLE
Is it possible to add embedded classes to a list?

### DISCUSSION THREAD

**Discussion by Skynse** [isar/isar#781] [Help] [answered]
I have a field in my `Task` class called `List<SubTask> subTasks = [];`

The embedded class is
```@embedded
class SubTask {
  String? title;
  bool completed = false;
}
```

When I try to append a subtask to that list, I get `Unsupported operation: Cannot add to a fixed-length list`

Any ideas?

**Comment by Skynse** [comment, marked as answer]
I think it's best that I use IsarLink. It took all day to find out that that existed.

edit: can confirm, this is the best approach.

**Comment by wieger** [comment]
You can actually make it work with embedded objects if you declare your list as a growable empty list.
`List<SubTask> subTasks = List.empty(growable: true);`

Side note: The Isar documentation explicitly advises to use embedded objects whenever possible see [Links](https://isar.dev/links.html) but of course it might still be better to stick with IsarLinks in your situation

**Reply by Skynse** [reply]
Using IsarLinks seems like a better approach. It's like having a Class and Students table and linking them instead of putting every single student in a list inside the Class table.

**Reply by MrCsabaToth** [reply]
I saw in the documentation that it advises an embedded list instead of Isar links, supposedly due to performance? Let's say if I go with embedding a list of objects, if I just have a `SubTask` in my hand, how do I access the embedder `Task`? In case of Isar Links there's a back link.

**Reply by m8811163008** [reply]
I prefer to stick to the documentation that states use embedded object. The work around perfectly fixes the issue of generating embedded list in a collection.
```dart
@collection
class FoodListCM {
  Id id = 0;

  List<FoodCM> foods = List.empty(growable: true);
}
@embedded
class FoodCM {
  late String name;

  int? calarie;
}
```

**Reply by Skynse** [reply]
Let's say I had a note taking app where I stored tags. Would an embedded object be a better choice? I've been using isarlinks but it feels so bad.

**Reply by agustincards14** [reply]
You should use IsarLinks if an object can exist without the other. In this case, tags exist without the notes, so use IsarLinks.

For the original case, you should use Embedded List because a SubTask is directly dependent on the Task.
